### DISCUSSION TITLE
APIs to download react Native application from EAS

### DISCUSSION THREAD

**Discussion by new-journey-05** [expo/expo#17124] [Archived: Questions]
I would like to see if there are any exposed APIs for EAS to download successfully generated build artefacts and run automation scripts based on the project & each successful build.

While the support to download manually from EAS portal is available and installation using the link provided, is there a way to  fetch the android / ios installer files based on parameters - project name, build status and timestamp ?

**Comment by fguitton** [comment]
Hello,

For a typical reactive CI pipeline EAS provides you with all you need in the response body of a build submission.
Run your EAS command with `--non-interactive --json` and parse the output. The download URL will be in `artifacts.buildUrl`.

If you want to do this arbitrarily you can query the GraphQL endpoint with the right access tokens in the same way `eas-cli` does it.
You can check the code out there: https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/graphql/queries/ProjectQuery.ts

🌴

**Reply by R4Y-R4Y** [reply]
can you give an example for the ci pipeline?
